network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -2470,2883 +454,534 @@ public void cloneNetwork(UUID uuid, int sourceVariantNum, int targetVariantNum,
-                        "calculatedBusesValid",
-                        "fictitious",
-                        SLACK_TERMINAL,
-                        ALIASES_WITHOUT_TYPE,
-                        ALIAS_BY_TYPE)
-                .whereColumn("networkUuid").isEqualTo(literal(networkUuid))
-                .whereColumn(VARIANT_NUM).isEqualTo(literal(variantNum))
-                .whereColumn("substationId").isEqualTo(literal(substationId))
-                .build())) {
-            List<Resource<VoltageLevelAttributes>> resources = new ArrayList<>();
-            for (Row row : resultSet) {
-                resources.add(Resource.voltageLevelBuilder()
-                        .id(row.getString(0))
-                        .variantNum(variantNum)
-                        .attributes(VoltageLevelAttributes.builder()
-                                .substationId(substationId)
-                                .name(row.getString(1))
-                                .properties(row.getMap(2, String.class, String.class))
-                                .nominalV(row.getDouble(3))
-                                .lowVoltageLimit(row.getDouble(4))
-                                .highVoltageLimit(row.getDouble(5))
-                                .topologyKind(TopologyKind.valueOf(row.getString(6)))
-                                .internalConnections(row.getList(7, InternalConnectionAttributes.class))
-                                .calculatedBusesForBusView(row.isNull(8) ? null : row.getList(8, CalculatedBusAttributes.class))
-                                .nodeToCalculatedBusForBusView(row.isNull(9) ? null : row.getMap(9, Integer.class, Integer.class))
-                                .busToCalculatedBusForBusView(row.isNull(10) ? null : row.getMap(10, String.class, Integer.class))
-                                .calculatedBusesForBusBreakerView(row.isNull(11) ? null : row.getList(11, CalculatedBusAttributes.class))
-                                .nodeToCalculatedBusForBusBreakerView(row.isNull(12) ? null : row.getMap(12, Integer.class, Integer.class))
-                                .busToCalculatedBusForBusBreakerView(row.isNull(13) ? null : row.getMap(13, String.class, Integer.class))
-                                .calculatedBusesValid(row.getBoolean(14))
-                                .fictitious(row.getBoolean(15))
-                                .slackTerminal(row.get(16, TerminalRefAttributes.class))
-                                .aliasesWithoutType(row.getSet(17, String.class))
-                                .aliasByType(row.getMap(18, String.class, String.class))
-                                .build())
-                        .build());
-            }
-            return resources;
-        }
+        return getEquipmentsInContainer(networkUuid, variantNum, substationId, SUBSTATION_ID, mappings.getVoltageLevelMappings(), VOLTAGE_LEVEL, Resource.voltageLevelBuilder(), VoltageLevelAttributes::new);
```

Comment:
I would use 'element' instead of 'equipment' because more general (a bus is not an equipment but a physical concept).

Reply:
Or maybe identifiable to be consistent with IIDM naming

Reply:
Renamed